pandas/core/generic.py

```diff
@@ -6524,6 +6524,11 @@ def fillna(
                         value, dtype_if_empty=object
                     )
                     value = value.reindex(self.index, copy=False)
+
```

Comment:
i think you are better off calling .align here. What you have on L6531 is too specific, fragile and trying to fix something after the fact.

Reply:
Yeah I see what you mean. Not seeing a clear solution with .align either - will have to have some sort of ugly isna check at some point I think to support this. Will close for now and reopen if I can find a better way to do this